---a/net/minecraft/network/play/client/C08PacketPlayerBlockPlacement.java
+++b/net/minecraft/network/play/client/C08PacketPlayerBlockPlacement.java
@@ -1,6 +1,7 @@
 package net.minecraft.network.play.client;
 
 import java.io.IOException;
+
 import net.minecraft.item.ItemStack;
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
@@ -8,8 +8,8 @@
 import net.minecraft.network.play.INetHandlerPlayServer;
 import net.minecraft.util.BlockPos;
 
-public class C08PacketPlayerBlockPlacement implements Packet
-{
+public class C08PacketPlayerBlockPlacement implements Packet {
+
     private static final BlockPos field_179726_a = new BlockPos(-1, -1, -1);
     private BlockPos position;
     private int placedBlockDirection;
@@ -19,14 +19,16 @@
     private float facingZ;
     private static final String __OBFID = "CL_00001371";
 
-    public C08PacketPlayerBlockPlacement() {}
+    public long timestamp; // CraftBukkit
 
-    public C08PacketPlayerBlockPlacement(ItemStack p_i45930_1_)
-    {
+    public C08PacketPlayerBlockPlacement() {
+    }
+
+    public C08PacketPlayerBlockPlacement(ItemStack p_i45930_1_) {
         this(field_179726_a, 255, p_i45930_1_, 0.0F, 0.0F, 0.0F);
     }
 
-    public C08PacketPlayerBlockPlacement(BlockPos p_i45931_1_, int p_i45931_2_, ItemStack p_i45931_3_, float p_i45931_4_, float p_i45931_5_, float p_i45931_6_)
-    {
+    public C08PacketPlayerBlockPlacement(BlockPos p_i45931_1_, int p_i45931_2_, ItemStack p_i45931_3_, float p_i45931_4_, float p_i45931_5_, float p_i45931_6_) {
         this.position = p_i45931_1_;
         this.placedBlockDirection = p_i45931_2_;
         this.stack = p_i45931_3_ != null ? p_i45931_3_.copy() : null;
@@ -36,64 +36,55 @@
         this.facingZ = p_i45931_6_;
     }
 
-    public void readPacketData(PacketBuffer buf) throws IOException
-    {
+    public void readPacketData(PacketBuffer buf) throws IOException {
+        timestamp = System.currentTimeMillis(); // CraftBukkit
         this.position = buf.readBlockPos();
         this.placedBlockDirection = buf.readUnsignedByte();
         this.stack = buf.readItemStackFromBuffer();
-        this.facingX = (float)buf.readUnsignedByte() / 16.0F;
-        this.facingY = (float)buf.readUnsignedByte() / 16.0F;
-        this.facingZ = (float)buf.readUnsignedByte() / 16.0F;
+        this.facingX = (float) buf.readUnsignedByte() / 16.0F;
+        this.facingY = (float) buf.readUnsignedByte() / 16.0F;
+        this.facingZ = (float) buf.readUnsignedByte() / 16.0F;
     }
 
-    public void writePacketData(PacketBuffer buf) throws IOException
-    {
+    public void writePacketData(PacketBuffer buf) throws IOException {
         buf.writeBlockPos(this.position);
         buf.writeByte(this.placedBlockDirection);
         buf.writeItemStackToBuffer(this.stack);
-        buf.writeByte((int)(this.facingX * 16.0F));
-        buf.writeByte((int)(this.facingY * 16.0F));
-        buf.writeByte((int)(this.facingZ * 16.0F));
+        buf.writeByte((int) (this.facingX * 16.0F));
+        buf.writeByte((int) (this.facingY * 16.0F));
+        buf.writeByte((int) (this.facingZ * 16.0F));
     }
 
-    public void processPacket(INetHandlerPlayServer p_180769_1_)
-    {
+    public void processPacket(INetHandlerPlayServer p_180769_1_) {
         p_180769_1_.processPlayerBlockPlacement(this);
     }
 
-    public BlockPos getPosition()
-    {
+    public BlockPos getPosition() {
         return this.position;
     }
 
-    public int getPlacedBlockDirection()
-    {
+    public int getPlacedBlockDirection() {
         return this.placedBlockDirection;
     }
 
-    public ItemStack getStack()
-    {
+    public ItemStack getStack() {
         return this.stack;
     }
 
-    public float getPlacedBlockOffsetX()
-    {
+    public float getPlacedBlockOffsetX() {
         return this.facingX;
     }
 
-    public float getPlacedBlockOffsetY()
-    {
+    public float getPlacedBlockOffsetY() {
         return this.facingY;
     }
 
-    public float getPlacedBlockOffsetZ()
-    {
+    public float getPlacedBlockOffsetZ() {
         return this.facingZ;
     }
 
-    public void processPacket(INetHandler handler)
-    {
-        this.processPacket((INetHandlerPlayServer)handler);
+    public void processPacket(INetHandler handler) {
+        this.processPacket((INetHandlerPlayServer) handler);
     }
-}
+
+    // SpigotMCP Start -- Remove as already fixed by mcp
+    /*
+    // CraftBukkit start - fix decompile error
+    @Override
+    public void processPacket(INetHandler pl) {
+        processPacket((PacketListenerPlayIn)pl);
+    }
+    // CraftBukkit end
+    */
+    // SpigotMCP End
+}
